File: python/sample.py
from datetime import datetime
import time

logtime_str = '20170101123456'
start = time.time()
for i in range(100000):
    dt = datetime.strptime(logtime_str, '%Y%m%d%H%M%S')

print('{:.4f}sec'.format(time.time() - start))

File: java/Sample.java
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.concurrent.TimeUnit;

public class Sample {

  public static void main(String[] args){
    String time = "20170101123456";
    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMddHHmmss");
    long start = System.currentTimeMillis();
    for(int i =0;i<100000;i++){
        LocalDateTime.parse(time, formatter);
    }
    System.out.println(((System.currentTimeMillis() - start)/1000d) + "sec");
  }
}
